### PR TITLE
Adds support for TTL in CosmosDB State Store

### PR DESCRIPTION
**Pull request by berndverst**
# Description

Adds support for TTL in CosmosDB State Store

## Issue reference
#306

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [x] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by berndverst** [comment]
Reference: https://docs.microsoft.com/en-us/azure/cosmos-db/time-to-live

File: state/azure/cosmosdb/cosmosdb.go [hunk 1]
```diff
@@ -422,3 +428,17 @@ func populatePartitionMetadata(key string, requestMetadata map[string]string) st
 
 	return key
 }
+
+func parseTTL(requestMetadata map[string]string) *int {
+	if val, found := requestMetadata[metadataTTLKey]; found && val != "" {
+		parsedVal, err := strconv.ParseInt(val, 10, 0)
+		if err != nil {
+			return nil
```
Comment (artursouza) [inline state/azure/cosmosdb/cosmosdb.go:436]: return error.

**Review by artursouza — CHANGES_REQUESTED** [review]

File: state/azure/cosmosdb/cosmosdb.go [hunk 2]
```diff
@@ -50,6 +51,7 @@ type CosmosItem struct {
 	Value        interface{} `json:"value"`
 	IsBinary     bool        `json:"isBinary"`
 	PartitionKey string      `json:"partitionKey"`
+	TTL          *int        `json:"ttl,omitempty"`
```
Comment (daixiang0) [inline state/azure/cosmosdb/cosmosdb.go:54]: wonder why it is a pointer? If for performance, why `PartitionKey` not?

Comment (artursouza) [reply state/azure/cosmosdb/cosmosdb.go:54]: This field is optional and there is a difference between 0 and nil TTL.

Comment (daixiang0) [reply state/azure/cosmosdb/cosmosdb.go:54]: I learn.

**Review by artursouza — APPROVED** [review]

**Comment by artursouza** [comment]
@berndverst Can you paste a screenshot on how the record in CosmosDB will look like?

**Pull request merged**